src/PointwiseFunctions/GeneralRelativity/Ricci.cpp

```diff
@@ -0,0 +1,61 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/Ricci.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+template <size_t SpatialDim, typename Frame, IndexType Index, typename DataType>
+tnsr::aa<DataType, SpatialDim, Frame, Index> compute_ricci(
+    const tnsr::abb<DataType, SpatialDim, Frame, Index>& christoffel_2nd_kind,
+    const tnsr::aBcc<DataType, SpatialDim, Frame, Index>&
+        d_christoffel_2nd_kind) {
+  auto ricci = make_with_value<tnsr::aa<DataType, SpatialDim, Frame, Index>>(
+      christoffel_2nd_kind, 0.);
+  constexpr auto dimensionality = index_dim<0>(ricci);
```

Comment:
`const` (that is apparently not redundant with constexpr)

Reply:
Can you please clarify what you are suggesting I change? Note this exact line appears e.g. on line 16 of christoffel.cpp; that's why I wrote it this way. Does christoffel.cpp also need to change?

Reply:
const constexpr auto ... 
and yes it should be changed similarly in Christoffel.cpp if it is const

Reply:
Sorry, I was apparently confused on this. a `constexpr` variable is `const`, however variables inside of a `constexpr` function need not be const. e.g.

```
constexpr int blah() {
  int a = 1;
  a++;
  return a;
}

static_assert(blah() == 2, "");
```

Experimentally it looks constexpr variables are always evaluated at compile time (this matches with what someone at the ISO meeting told me). There does appear to be a recursion depth limit, on GCC 7.2 and clang 5 it's 512.

Sorry for my confusion!

Reply:
Fixed: I made dimensionality const constexpr auto ... in Ricci.cpp and Christoffel.cpp